js/tests/enhancedDate.spec.js

```diff
@@ -0,0 +1,79 @@
+describe("Enhanced date", function () {
+	
+
+	describe("getDayName", function () {
+		before(function() {
+			enhancedDate.setDate(new Date("05-23-2015"));
+		});
+		it("should return Saturday", function() {
+			expect(enhancedDate.getDayName()).to.be.equal("Saturday");
+		});
+		it("should be a string", function() {
+			expect(enhancedDate.getDayName()).to.be.a("string");
+		});
+	});
+
+	describe("setDate", function() {		
+
+		it("should return instanceof date", function () {
+        	expect(enhancedDate.setDate(new Date("05-23-2015"))).to.be.an.instanceof(Date);        	
+		});
+		it("should execute fast", function (done) {
+        	expect(enhancedDate.setDate(new Date("05-23-2015"))).to.be.an.instanceof(Date); 
```

Comment:
This still has the same problem as before - it's the same test. There's no need to do a "timing" test for operations that are not asynchronous.